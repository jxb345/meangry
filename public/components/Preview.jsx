import React, { useState } from 'react';

const Preview = (props) => {
  console.log('props', props);
  return (
    <div>
      {
        (!props.emailSent)
        ?
        <div className="preview">
          <div className="subject-preview">
            {props.subject}
          </div>
          <br/>
          <div className="body-preview">
            {props.body}
          </div>
        </div>
        :
        <div className="sent-preview">
          Feeling any better? <br/>
          Your heatMail has been sent.
        </div>
      }
    </div>
  )
}

export default Preview;