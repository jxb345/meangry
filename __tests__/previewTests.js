import React from 'react';
import renderer from 'react-test-renderer';
import Preview from '../public/components/Preview.jsx';

test('preview-snapshot-test', () => {
  const component = renderer.create(
    <Preview body='body' subject='subject'/>);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    console.log('tree------c.p', tree.children[0].props);
})