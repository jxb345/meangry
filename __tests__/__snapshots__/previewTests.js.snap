// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preview-snapshot-test 1`] = `
<div>
  <div
    className="preview"
  >
    <div
      className="subject-preview"
    >
      subject
    </div>
    <div
      className="body-preview"
    >
      body
    </div>
  </div>
</div>
`;

exports[`test 1`] = `
<div>
  <div
    className="preview"
  >
    <div
      className="subject-preview"
    >
      subject
    </div>
    <div
      className="body-preview"
    >
      body
    </div>
  </div>
</div>
`;
